flink-ml-core/src/main/java/org/apache/flink/ml/util/ReadWriteUtils.java

```diff
@@ -186,7 +174,7 @@ private static String getDataPath(String path) {
     private static String getPathForPipelineStage(int stageIdx, int numStages, String parentPath) {
         String format = String.format("%%0%dd", String.valueOf(numStages).length());
         String fileName = String.format(format, stageIdx);
-        return Paths.get(parentPath, "stages", fileName).toString();
+        return new Path(parentPath, new Path("stages", fileName)).toString();
```

Comment:
I understand that `Paths.get()` might not work in hdfs and thus should be replaced. Do we have an even better choice than `new Path(parentPath, new Path("stages", fileName))`? It seems that we would need N-1 `new Path()` in order to join N strings, which might not be good enough.